Rakefile

```diff
@@ -1,15 +1,49 @@
-require 'bundler/gem_tasks'
 require 'rubocop/rake_task'
 require 'rake/testtask'
+require 'rspec/core/rake_task'
 
 # Test task is not part of default task list,
 # because it requires a node to test against.
-task default: %w(rubocop build)
+task default: %w(rubocop spec build)
 
 RuboCop::RakeTask.new
 
+# Because each of the below specs requires a clean Ruby environment,
+# they need to be run individually instead of as a single RSpec task.
+RSpec::Core::RakeTask.new(:spec_no_clients) do |t|
+  t.pattern = 'spec/no_clients_spec.rb'
+end
+RSpec::Core::RakeTask.new(:spec_nxapi_only) do |t|
+  t.pattern = 'spec/nxapi_only_spec.rb'
+end
+RSpec::Core::RakeTask.new(:spec_grpc_only) do |t|
+  t.pattern = 'spec/grpc_only_spec.rb'
+end
+RSpec::Core::RakeTask.new(:spec_all_clients) do |t|
+  t.pattern = 'spec/all_clients_spec.rb'
+end
+
+task spec: [:spec_no_clients,
+            :spec_nxapi_only,
+            :spec_grpc_only,
+            :spec_all_clients,
+           ]
+
 task :build do
-  system 'gem build cisco_os_shim.gemspec'
+  Dir['*.gemspec'].each { |gemspec| system "gem build #{gemspec}" }
+  # Validate gem contents
+  if system('gem specification ./*nxapi*gem | grep -v homepage | grep grpc')
```

Comment:
Just a sanity check - proving to myself that I've correctly split the gems apart so that 'cisco_os_shim.gem' doesn't contain gRPC or NXAPI code, 'cisco_os_shim-nxapi.gem' doesn't contain gRPC, and 'cisco_os_shim-grpc.gem' doesn't contain NXAPI.